File: ocaml/fizzbuzz.ml
let fizzBuzz i =
  let wichFizzBuzz x = match x with
    | x when ((x mod 3 == 0) && (x mod 5 == 0)) -> "FizzBuzz"
    | x when (x mod 3 == 0)			-> "Fizz"
    | x when (x mod 5 == 0) 		    	-> "Buzz"
    | _		 			    	-> string_of_int x
  in let rec fizzBuzzRec x =
      Printf.printf " %s" (wichFizzBuzz x);
      if x != i then fizzBuzzRec (x+1)
    in Printf.printf "1";
      fizzBuzzRec 2;
      Printf.printf "\n"
;;

fizzBuzz 100;

File: go/fizzbuzz.go
package main

import "fmt"

func main() {
	for i := 1; i < 100; i++ {
		if i % 3 == 0 {
			if i % 5 == 0 {
				fmt.Printf("FizzBuzz")
			} else {
				fmt.Printf("Fizz")
			}
		} else if i % 5 == 0 {
			fmt.Printf("Buzz")
		} else {
			fmt.Printf("%d", i)
		}
	}
}
